src/lib.rs

```diff
@@ -189,6 +189,9 @@
 
 #![warn(missing_docs)]
 
+/// Map implementation used by the library
+pub type Map<K, V> = rustc_hash::FxHashMap<K, V>;
```

Comment:
This may be subtle, but since I've been coding in Elm, I always avoid dependency cycles in modules. Putting the type alias here in `lib.rs` and importing it in submodules create a dependency cycle. Rust is smart enough to solve it but if possible I'd like to avoid it.

Reply:
could we make a `mod type_aliases` that has this one line, and is imported by all the places that use it?

Reply:
that would be my preference indeed

Reply:
I thought of putting it in a module but for the reasons of `lib.rs` containing only module declarations and looking clean. I didn't think of the dependency cycle.

Would you like it being exported from the root as it is now? I think `pubgrub::Map` would be more convenient for users of the library than `pubgrub::type_aliases::Map`.
If we export, the module shouldn't be made `pub` and instead placed in `internal` I think.

And about the naming:
1) `type_aliases`
2) `types`
3) `exports`
4) something else?

Reply:
I moved it to `internal::types` and exported that from root. Naming is subject for discussion. Please take a look.